metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java

```diff
@@ -134,29 +135,28 @@ public PartitionServiceImpl(
         @Nullable final Pageable pageable,
         final boolean includeUserDefinitionMetadata,
         final boolean includeUserDataMetadata,
-        final GetPartitionsRequestDto getPartitionsRequestDto
+        @Nullable final GetPartitionsRequestDto getPartitionsRequestDto
     ) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+        final ConnectorPartitionService service = connectorManager.getPartitionService(name.getCatalogName());
+        // the conversion will handle getPartitionsRequestDto as null case
+        final PartitionListRequest partitionListRequest =
+            converterUtil.toPartitionListRequest(getPartitionsRequestDto, pageable, sort);
 
-        String filterExpression = null;
-        List<String> partitionNames = null;
-        if (getPartitionsRequestDto != null) {
-            filterExpression = getPartitionsRequestDto.getFilter();
-            partitionNames = getPartitionsRequestDto.getPartitionNames();
-        }
+        final ConnectorRequestContext connectorRequestContext = converterUtil.toConnectorContext(metacatRequestContext);
+        final List<PartitionInfo> resultInfo = service
```

Comment:
This should come after the validation check that throws IllegalArugumentException.

Reply:
Sounds right.